weave

```diff
@@ -145,9 +177,11 @@ collect_ip_args() {
 if [ "$1" = "run" ] ; then
     shift 1
     check_docker_version
-    if [ "$1" = "--with-dns" ] ; then
-        shift 1
-        DNS_ARGS=$(exec_remote dns-args "$@")
+    collect_dns_args "$@"
```

Comment:
This has the exact same semantics as `collect_cidr_args` - it consumes zero or more dns related options from the _start_ of `$@`, stopping as soon as it hits the first thing that isn't...